File: dynamic/src/main/java/com/github/rviannaoliveira/dynamic/domain/model/DynamicActionProperties.kt
package com.github.rviannaoliveira.dynamic.domain.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class DynamicActionProperties (
    @JsonProperty("deeplink")
    val deeplink : String?,

    @JsonProperty("actionData")
    val actionData: Any?,

    @JsonProperty("analytics")
    val analytics: DynamicAnalyticsProperties?,
)

File: dynamic-flutter/flutter_dynamics_module/lib/data/model/action/dynamic_action_properties.dart
import 'dynamic_analytics_properties.dart';

class DynamicActionProperties {
  final String? deeplink;
  final DynamicAnalyticsProperties? analyticsProperties;

  const DynamicActionProperties({this.deeplink, this.analyticsProperties});

  factory DynamicActionProperties.fromJson(dynamic? properties) {
    return DynamicActionProperties(
        deeplink: properties != null ? properties["deeplink"] : null,
        analyticsProperties: properties != null ? DynamicAnalyticsProperties
            .fromJson(
            properties[DynamicAnalyticsProperties.key]) : null);
  }

  static const String key = "actionProperties";
}